src/rdkafka_subscription.c

```diff
@@ -112,6 +112,16 @@ rd_kafka_assign (rd_kafka_t *rk,
 
 
 
+int
+rd_kafka_assignment_lost (rd_kafka_t *rk) {
+        rd_kafka_cgrp_t *rkcg;
+
+        if (!(rkcg = rd_kafka_cgrp_get(rk)))
+                return 0;
+
+        return rkcg->rkcg_assignment_lost ? 1 : 0;
```

Comment:
All rkcg_assignment_lost accesses needs locking, or changing it to an atomic32 (which is easier), since this is called from an application thread, while rkcg_assignment_lost is set from the rdkafka main thread,